dolphinscheduler-api/src/test/java/org/apache/dolphinscheduler/api/service/LoggerServiceTest.java

```diff
@@ -272,9 +312,28 @@ public void testGetLogBytesInSpecifiedProject() {
         taskInstance.setHost("127.0.0.1:" + nettyServerPort);
         taskInstance.setLogPath("/temp/log");
         doNothing().when(projectService).checkProjectAndAuthThrowException(loginUser, projectCode, DOWNLOAD_LOG);
+
+        when(taskInstanceDao.queryById(1)).thenReturn(null);
+        assertThrowsServiceException(
+                Status.INTERNAL_SERVER_ERROR_ARGS, () -> loggerService.getLogBytes(loginUser, projectCode, 1));
+
         when(taskInstanceDao.queryById(1)).thenReturn(taskInstance);
         when(taskDefinitionMapper.queryByCode(taskInstance.getTaskCode())).thenReturn(taskDefinition);
         assertDoesNotThrow(() -> loggerService.getLogBytes(loginUser, projectCode, 1));
+
+        taskDefinition.setProjectCode(2L);
+        assertThrowsServiceException(Status.INTERNAL_SERVER_ERROR_ARGS,
+                () -> loggerService.getLogBytes(loginUser, projectCode, 1));
+        taskDefinition.setProjectCode(1L);
+
+        try (
+                MockedStatic<SingletonJdkDynamicRpcClientProxyFactory> ignored =
+                        Mockito.mockStatic(SingletonJdkDynamicRpcClientProxyFactory.class)) {
+            when(SingletonJdkDynamicRpcClientProxyFactory.getProxyClient(any(), any())).thenReturn(iLogService);
+            when(iLogService.pageQueryTaskInstanceLog(any())).thenThrow(ServiceException.class);
+            assertThrowsServiceException(Status.DOWNLOAD_TASK_INSTANCE_LOG_FILE_ERROR,
+                    () -> loggerService.getLogBytes(loginUser, projectCode, 1));
+        }
```

Comment:
If only adding this kind of bad case is not helpful, can we add a log server provider?

Reply:
just missed the point, i'll make the change

Reply:
> If only adding this kind of bad case is not helpful, can we add a log server provider?

updated